### ISSUE TITLE
Can you please add sample CSV file?

### ISSUE THREAD

**Issue by cricrazy** [binjr/binjr#87]
I have some time series CSV files with time stamp. But I cannot load it in. Can some one please add a sample CSV file to test it on?

**Comment by fthevenet** [comment]
Hi,
I have added a couple of csv file samples here:
https://binjr.eu/samples/csv/windiskiobytes-_Total.csv
https://binjr.eu/samples/csv/windiskioop-_Total.csv
https://binjr.eu/samples/csv/windiskiopercent-_Total.csv

If you're having trouble opening your own csv files with binjr, please check the following:
* The timestamps must be carried by the _first_ column of the csv file
* Make sure that the encoding, delimiter and parsing format for the timestamp match the csv file you're trying to open.
* The syntax for date parsing pattern is that of Java's DatetimeFormatter: https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
* You can consult the application's logs by pressing the F12 key; a more detailed reason as to why binjr refuses to open the file should show there.

**Comment by cricrazy** [comment]
@fthevenet Thank you. Really appreciate it. I found several issues with my CSV, even though I had a pretty standard time stamp in my time series data. Here are some of the observations.

- It somehow does not like carriage return and line feed. 
- I keep getting this NaN for min, max, avg, without a helpful message. 

I am trying to get this to plot without any success. 

```
Date,Temp,CO,SO2
2019-10-04 10:13:00,28.8,1.1,0.1
2019-10-04 10:14:00,29.6,2.1,0.1
2019-10-04 10:15:00,29.4,0.6,0.1
2019-10-04 10:16:00,33.6,0.2,0.1
2019-10-04 10:17:00,36.9,0.1,0.1
2019-10-04 10:18:00,37.9,0.3,0.1
2019-10-04 10:19:00,37.5,0.2,0.1
2019-10-04 10:20:00,36.8,0.5,0.1
2019-10-04 10:21:00,41.4,0.8,0.1
2019-10-04 10:22:00,45.0,0.4,0.1
2019-10-04 10:23:00,50.9,0.2,0.1
2019-10-04 10:24:00,51.9,0.2,0.1
2019-10-04 10:25:00,51.6,0.1,0.1
```

Can you help? Is there a way to skip the Date/Time Stamp and instead add an Index as the first column?

**Comment by fthevenet** [comment]
OK, I think the parsing of the file is actually working fine; the fact that you see no data on the graph is most likely due to the fact that there are no data for the time interval you're currently looking at.

By default binjr set the last 24 hours from the current instant as the displayed time range; if there are no samples for that period, nothing is graphed  and min, max and avg and set to NaN (Not a Number). But since this isn't considered an error either, no message is displayed.

If you change the time range to include the timestamps you've got data for (from 2019-10-04 10:13:00 to 2019-10-04 10:25:00 in the example above) you should see something.

**Comment by cricrazy** [comment]
Oh Wow....I spent quite a bit of time trying figure it out. Wish it is documented somewhere. 

I did not realize the top button that let's me select the date range. 

![image](https://user-images.githubusercontent.com/4622816/66683843-17076980-ec3e-11e9-94c6-a71563b37f9f.png)

Is it possible to have an Index, rather than a time and date stamp as first column?

**Comment by fthevenet** [comment]
Admittedly, the doc is quite incomplete, and what's actually in it is sometime obsolete.
Fixing it is on the TODO list.... 

To answer your question; no it is not possible, binjr only supports dates and times for X axis coordinates.

**Comment by tstuefe** [comment]
Christ guys that button is really hidden in plain sight :) Took me minutes to find it, and I only tried because I saw the screenshot from @cricrazy above. Why not set the range automatically to the first/last sample? I would expect that is the point where many folks give up, which is a shame considering the tool is good.

**Comment by fthevenet** [comment]
Well, I admit there must be a real UX issue here since you're the second person to bump into it...

As a matter of fact, following that ticket a feature was added to automatically adjust the time range  when you drop series from a csv file so that it creates a new tab (i.e. if you drop it onto the empty stage or on the icon at the top of a worksheet, see attached screen capture).

![dnd](https://user-images.githubusercontent.com/7450507/70343734-575b2080-1858-11ea-93e1-b38efc801c80.gif)

The time range is **not** changed if you drop it onto an existing worksheet (i.e. an existing tab with a time range already defined), however since it would affect the typical workflow of how the tool is used internally, where we mixed series from multiple sources onto the same graphs and having the time range reset every time a series is added would be counterproductive.

In hindsight, automatically adjusting the time range up until there is at least one series onto a tab could be a good compromise.

As for the time range button, I though it was big enough, but again I must admit that it is probably not as obvious as I thought. I clearly blame the current "flat design" trend... (let us ignore the fact that I succumbed to it, shall we ;-P )

**Comment by tstuefe** [comment]
Oh I see. When I drop a column onto the x+ symbol, it creates the worksheet and adjusts the time line accordingly. I did not know that was possible, so I created an empty worksheet and dropped the values onto that one.

Question though, I multi selected three columns and dropped them, only to get three charts. But what if I want them in one chart (which I do)? Is there a way to merge the charts into one?

Alternatively, maybe ask when doing the drop whether the user wants them all in one charts or separated? Maybe with a "Don't ask again" checkbox.

>In hindsight, automatically adjusting the time range up until there is at least one series onto a tab >could be a good compromise.

Yes, that I would have expected.

>As for the time range button, I though it was big enough, but again I must admit that it is probably >not as obvious as I thought. I clearly blame the current "flat design" trend... (let us ignore the fact >that I succumbed to it, shall we ;-P )

I like the material flat design. Looks good. As for the time range, I randomly clicked onto the x axis below the chart because I somehow expected that to do something, to no avail. The header line as a button is just unexpected.

Thank you!

**Comment by fthevenet** [comment]
Hi,

I've been thinking about the time range picker control not being obviously spotted by first time users, as I fully agree that this is a big UI fail with the potential to frustrate potential users away.

I've added a small downward-pointing arrow to the right of label, like for a drop down list control, which hopefully should hint that clicking on the label will reveal more options.
Also, I've implemented some of the ideas in #65 (the step back/forward arrows) so altogether I'm hoping for this control to have an increased "click me" feel, and generally better discoverability, without cluttering the top bar too much.

![screendump](https://user-images.githubusercontent.com/7450507/70514452-f7a09600-1b33-11ea-9753-729d52eee78e.png)

All feedback very welcome! (@Arkanosis, if you could chime in too)
(You can also try it live right now  by checking out the timerangectrl branch and runnng `gradlew run`, though be warned that it is a very volatile feature branch)

**Comment by fthevenet** [comment]
> Question though, I multi selected three columns and dropped them, only to get three charts. But what if I want them in one chart (which I do)? Is there a way to merge the charts into one?

There is no feature to merge all series for different chart into one, at least not per se.
There is however a feature that lets you see several independent graphs overlaid on the same chart area.
The difference is that each graphs keeps its own Y axis, type and display properties; this is useful when you need want to compare series with completely different units (e.g. seconds and MB) to see how one might affect the other.

![binjr2](https://user-images.githubusercontent.com/7450507/70517147-6a137500-1b38-11ea-8f72-405070e2f62f.gif)

This might not be a good fit for your use case though (the graph become difficult to read when you have many series overlaid), but still it's probably good to know it exists.
